addons/event/static/src/client_action/event_barcode.js

```diff
@@ -64,7 +64,7 @@ export class EventScanView extends Component {
      * formatted registration information, notably its status or event-related
      * information. Open a confirmation / choice Dialog to confirm attendee.
      */
-    async onBarcodeScanned(barcode) {
+    async onBarcodeScanned(barcode, source) {
```

Comment:
give reader as a default value over here so don't need to check it later in the `doNextScan` function